// mathjs/lib/cjs/entry/dependenciesAny/dependenciesSymbolicEqual.generated.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.symbolicEqualDependencies = void 0;
var _dependenciesOperatorNodeGenerated = require("./dependenciesOperatorNode.generated.js");
var _dependenciesParseGenerated = require("./dependenciesParse.generated.js");
var _dependenciesSimplifyGenerated = require("./dependenciesSimplify.generated.js");
var _dependenciesTypedGenerated = require("./dependenciesTyped.generated.js");
var _factoriesAny = require("../../factoriesAny.js");
/**
 * THIS FILE IS AUTO-GENERATED
 * DON'T MAKE CHANGES HERE
 */

var symbolicEqualDependencies = exports.symbolicEqualDependencies = {
  OperatorNodeDependencies: _dependenciesOperatorNodeGenerated.OperatorNodeDependencies,
  parseDependencies: _dependenciesParseGenerated.parseDependencies,
  simplifyDependencies: _dependenciesSimplifyGenerated.simplifyDependencies,
  typedDependencies: _dependenciesTypedGenerated.typedDependencies,
  createSymbolicEqual: _factoriesAny.createSymbolicEqual
};
